float32 dist
int32 map_num
---